WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormRemoteActionUseCase.swift

```diff
@@ -205,4 +258,89 @@ private extension ProductFormRemoteActionUseCase {
         }
         stores.dispatch(passwordUpdateAction)
     }
+
+    func duplicateVariations(_ variationIDs: [Int64],
+                             from oldProductID: Int64,
+                             to newProduct: EditableProductModel,
+                             onCompletion: @escaping (Result<EditableProductModel, ProductUpdateError>) -> Void) {
+        Task { [weak self] in
+            guard let self = self else { return }
+            for id in variationIDs {
+                guard let variation = await self.retrieveProductVariation(id: id, siteID: newProduct.siteID, productID: oldProductID) else {
+                    continue
+                }
+                let newVariation = CreateProductVariation(regularPrice: variation.regularPrice ?? "", attributes: variation.attributes)
+                await self.duplicateProductVariation(newVariation, parent: newProduct)
+            }
```

Comment:
I think we can send these requests in parallel by using a `TaskGroup` instead of waiting for each variation to finish before duplicating the next variation. 

We can do something like the following,
```suggestion
            await withTaskGroup(of: Void.self) { group in
                for id in variationIDs {
                    group.addTask {
                        guard let variation = await self.retrieveProductVariation(id: id, siteID: newProduct.siteID, productID: oldProductID) else {
                            return
                        }
                        let newVariation = CreateProductVariation(regularPrice: variation.regularPrice ?? "", attributes: variation.attributes)
                        await self.duplicateProductVariation(newVariation, parent: newProduct)
                    }
                }
            }
```

What do you think?

Reply:
Great thinking! 💯  Duplicating variations concurrently make the process faster - I updated the code following this in 6520428c752093953ad4dabcf49528fc7d161cfe.